src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -222,51 +223,242 @@ private bool ScanDesignation(bool permitTuple)
+                bool parsePropertySubpattern()
+                {
+                    if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken)
+                    {
+                        propertySubpattern = ParsePropertySubpattern();
+                        return true;
+                    }
+
+                    return false;
+                }
+
+                VariableDesignationSyntax designation = null;
+                bool parseDesignation()
+                {
+                    if (this.IsTrueIdentifier() && (!forCase || this.CurrentToken.ContextualKind != SyntaxKind.WhenKeyword))
+                    {
+                        designation = ParseSimpleDesignation();
+                        return true;
+                    }
+
+                    return false;
+                }
+
+                bool looksLikeCast()
+                {
+                    var resetPoint2 = this.GetResetPoint();
+                    try
+                    {
+                        return this.ScanCast(forPattern: true);
                     }
-                    if (node == null)
+                    finally
                     {
-                        // it is an expression for typical switch case. 
+                        this.Reset(ref resetPoint2);
+                        this.Release(ref resetPoint2);
+                    }
+                }
+
+                if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken && (type != null || !looksLikeCast()))
+                {
+                    // It is possible this is a parenthesized (constant) expression.
+                    // We normalize later.
+                    ParseSubpatternList(out var openParenToken, out var subPatterns, out var closeParenToken, SyntaxKind.OpenParenToken, SyntaxKind.CloseParenToken);
+                    if (this.CurrentToken.Kind == SyntaxKind.EqualsGreaterThanToken)
+                    {
+                        // Testers do the darndest things, in this case putting a lambda expression where a pattern is expected.
                         this.Reset(ref resetPoint);
-                        node = this.ParseSubExpression(Precedence.Expression);
+                        return this.ParseSubExpression(Precedence.Expression);
+                    }
+
+                    parsePropertySubpattern();
+                    parseDesignation();
+
+                    if (type == null &&
+                        propertySubpattern == null &&
+                        designation == null &&
+                        subPatterns.Count == 1 &&
+                        subPatterns[0].NameColon == null &&
+                        subPatterns[0].Pattern is ConstantPatternSyntax cp)
+                    {
+                        // There is an ambiguity between a deconstruction pattern `(` pattern `)`
+                        // and a constant expression pattern than happens to be parenthesized.
```

Comment:
>than [](start = 61, length = 4)

Typo?

Reply:
Recorded in https://github.com/dotnet/roslyn/issues/23222.